### PR TITLE
added front room creating. added server logic in socket master and ch…

### PR DESCRIPTION
**Pull request by SkrypnikMR**
…at master. aded modal on front, aded saga for creating new room, added new listeners for socket on front

### CODE REVIEW

**Review by Rashevskyyy — APPROVED** [review]

File: client/src/components/Chat/ChatList/ChatList.jsx [hunk 1]
```diff
@@ -1,26 +1,44 @@
...
+      <StCreateRoom>
+        <Button
+          id='create_room'
+          name='create_room'
+          onClick={handleCreateRoom}
+          content={t('create_room')}
+          width="200px"
+        />
+      </StCreateRoom>
+
```
Comment (MaxShvachko) [inline client/src/components/Chat/ChatList/ChatList.jsx]: empty line

File: server/Classes/ChatManager.js [hunk 2]
```diff
@@ -55,6 +66,37 @@ class ChatManager {
...
+    }
+    readAllMessages = async ({ body }, res) => {
+        try {
+            const { user_id, room_id, lastMessageTime } = body;
+            const check = await this.connect.query(`SELECT * FROM unread_state where room_id = ${room_id} AND user_id = ${user_id}`);
+            if (check.length > 0) {
+                await this.connect.query(`UPDATE unread_state SET last_read_date = '${moment(lastMessageTime).format('YYYY-MM-DD HH:mm:ss')}' where room_id = ${room_id} AND user_id = ${user_id}`)
+                return res.status(200).json({ message: 'done' });
+            }
+            await this.connect.query(`INSERT INTO unread_state (room_id, user_id, last_read_date) VALUES('${room_id}', '${user_id}', '${moment(lastMessageTime).format('YYYY-MM-DD HH:mm:ss')}');`)
```
Comment (MaxShvachko) [inline server/Classes/ChatManager.js]: вынеси преобразование в moment в хелпер, много дубляжа

File: client/src/components/Chat/ChatList/ChatListItems/ChatListItems.jsx [hunk 3]
```diff
@@ -15,19 +28,23 @@ const ChatListItems = ({ color, img, content, setValue, id }) => {
       <StPhoto>
         <img src={img ? state.src : state.defaultImg} onError={onError} />
       </StPhoto>
-      <div>
-        {content}
-      </div>
+      <StRoom>
+        <p>{content}</p>
+        {unreadCount === 0 ? null : <span>{unreadCount}</span>}
```
Comment (MaxShvachko) [inline client/src/components/Chat/ChatList/ChatListItems/ChatListItems.jsx]: unreadCount ? <span>{unreadCount}</span> : null

File: client/src/components/Chat/ChatContent/ChatDisplay/ChatDisplay.jsx [hunk 4]
```diff
@@ -13,13 +17,14 @@ const ChatDisplay = ({ messages, currentUser, currentRoomName }) => {
             author={message.author}
             key={message.time}
             messageText={message.text}
-            messageTime={support.getPrettyTime(message.time)}
+            messageTime={moment(message.time).format('YYYY-MM-DD HH:mm:ss')}
```
Comment (MaxShvachko) [inline client/src/components/Chat/ChatContent/ChatDisplay/ChatDisplay.jsx]: в хелпер moment.format()

File: client/src/Store/chat/saga.js [hunk 5]
```diff
@@ -85,7 +90,7 @@ export function* sendMessageSaga() {
             text: message,
             room_name,
             room_id,
-            time: new Date().getTime(),
+            time: moment().format('YYYY-MM-DD HH:mm:ss'),
```
Comment (MaxShvachko) [inline client/src/Store/chat/saga.js]: и тут замени на хелпер

**Pull request merged**